### PR TITLE
chore: add test/util/genesis on v1.x

### PR DESCRIPTION
**Pull request by ninabarbakadze**
## Overview

adding genesis to utils as part of #2414

### CODE REVIEW

**Review by rach-id — COMMENTED** [review]
As a first pass, why is this needed to be in v1.x?

**Comment by ninabarbakadze** [comment]
> As a first pass, why is this needed to be in v1.x?

non determinism unit test. testApp needs to be constructed the same way so i wanted to reuse this logic.

File: test/util/testnode/config.go [hunk 1]
```diff
@@ -7,42 +7,37 @@ import (
 	"github.com/celestiaorg/celestia-app/cmd/celestia-appd/cmd"
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
 	v1 "github.com/celestiaorg/celestia-app/pkg/appconsts/v1"
+	"github.com/celestiaorg/celestia-app/test/util/genesis"
 	"github.com/cosmos/cosmos-sdk/server"
 	srvconfig "github.com/cosmos/cosmos-sdk/server/config"
 	srvtypes "github.com/cosmos/cosmos-sdk/server/types"
 	tmconfig "github.com/tendermint/tendermint/config"
-	tmrand "github.com/tendermint/tendermint/libs/rand"
+	// tmrand "github.com/tendermint/tendermint/libs/rand"
```
Comment (rootulp) [inline test/util/testnode/config.go:15]:
```suggestion
```

File: app/test/integration_test.go [hunk 2]
```diff
@@ -206,78 +201,6 @@ func (s *IntegrationTestSuite) TestMaxBlockSize() {
 	}
 }
 
-func (s *IntegrationTestSuite) TestSubmitPayForBlob() {
```
Comment (rootulp) [inline app/test/integration_test.go:209]: [question] was this test moved or just deleted? It doesn't look like the original PR deleted them: https://github.com/celestiaorg/celestia-app/pull/2455/files#diff-d66f948da147c1b292dcf8131bd525c02c6e23daa6c64712a85e53476271cc65L60

File: app/test/integration_test.go [hunk 3]
```diff
@@ -385,66 +308,6 @@ func (s *IntegrationTestSuite) TestEmptyBlock() {
 	}
 }
 
-// TestSubmitPayForBlob_blobSizes verifies the tx response ABCI code when
-// SubmitPayForBlob is invoked with different blob sizes.
-func (s *IntegrationTestSuite) TestSubmitPayForBlob_blobSizes() {
```
Comment (rootulp) [inline app/test/integration_test.go:390]: [question] was this test moved or just deleted? It doesn't look like the original PR deleted them: https://github.com/celestiaorg/celestia-app/pull/2455/files#diff-d66f948da147c1b292dcf8131bd525c02c6e23daa6c64712a85e53476271cc65L60

**Review by rootulp — APPROVED** [review]
LGTM. One question about whether tests were moved or deleted.

File: app/test/integration_test.go (hunk 2 above)
Comment (ninabarbakadze) [reply app/test/integration_test.go:209]: it was deleted like in this pr https://github.com/celestiaorg/celestia-app/pull/2834/files#diff-d66f948da147c1b292dcf8131bd525c02c6e23daa6c64712a85e53476271cc65

File: app/test/integration_test.go (hunk 3 above)
Comment (ninabarbakadze) [reply app/test/integration_test.go:390]: ditto

File: app/test/integration_test.go (hunk 2 above)
Comment (rootulp) [reply app/test/integration_test.go:209]: Thanks. Resolving

**Review by rootulp — APPROVED** [review]

**Review by cmwaters — CHANGES_REQUESTED** [review]
There's a lot of go API breaking changes in this PR. If you want to add the genesis package, it needs to be a purely additive change without removing several files and functions.

**Review by rach-id — APPROVED** [review]
utAck

**Comment by rach-id** [comment]
> There's a lot of go API breaking changes in this PR

Should we worry about breaking changes to the testing infrastructure in release branches?

**Review by evan-forbes — APPROVED** [review]

**Comment by cmwaters** [comment]
> Should we worry about breaking changes to the testing infrastructure in release branches?

**Comment by rootulp** [comment]
I revise my review b/c the testing infra is exported so this is technically API breaking like Callum mentioned. 

@ninabarbakadze is it possible to add new exported functions and deprecate existing functions to avoid all API breaks?

**Comment by ninabarbakadze** [comment]
> I revise my review b/c the testing infra is exported so this is technically API breaking like Callum mentioned.
> 
> @ninabarbakadze is it possible to add new exported functions and deprecate existing functions to avoid all API breaks?

Yes, I'm going to manually add the genesis folder without cherry picking evan's commit and see if it'll make the process of maintaining the API easier.

**Review by rootulp — APPROVED** [review]

**Review by cmwaters — APPROVED** [review]

**Pull request merged**